source/core/src/main/com/csse3200/game/areas/terrain/CropTileComponent.java

```diff
@@ -0,0 +1,111 @@
+package com.csse3200.game.areas.terrain;
+
+import com.badlogic.gdx.math.MathUtils;
+import com.csse3200.game.components.Component;
+import com.csse3200.game.services.ServiceLocator;
+
+/**
+ * Component which stores information about plots of land on which crops and plants can grow.
+ * Listens to a variety of trigger-able events to update state of the tile, including its water
+ * content and whether it is fertilised.
+ */
+public class CropTileComponent extends Component {
+
+    private final float waterDecreaseRate = 0.05f;
+
+    private final float idealWaterFalloffSharpness = 2.0f;
+    private final float idealWaterFalloffTolerance = 1.2f;
+    private final float waterDamageThreshold = 0.1f;
+
+    private float waterContent;
+    private final float soilQuality;
+    private boolean isFertilised;
+    private boolean isOccupied;
+
+    /**
+     * Creates a new crop tile component, with a specified initial water content, and an initial soil quality (higher values indicate higher soil
+     * quality, meaning better growth rate).
+     *
+     * @param initialWaterContent The initial amount of water in the crop tile. Value should be
+     *                            between 0 and 2 inclusive - 0 indicates tile is completely dry,
+     *                            and 2 indicates tile is flooded, water content of 1 leads to ideal
+     *                            growth rate for most plants
+     * @param soilQuality The initial soil quality of the crop tile. Higher soil quality increases
+     *                    the calculated growth rate. Should be greater than 0.
+     */
+    public CropTileComponent(float initialWaterContent, float soilQuality) {
+        this.waterContent = initialWaterContent;
+        this.soilQuality = soilQuality;
+        this.isFertilised = false;
+        this.isOccupied = false;
+    }
+
+    /**
+     * Adds player-triggered events listeners to the entity.
+     */
+    @Override
+    public void create() {
+        entity.getEvents().addListener("water", this::waterTile);
+        entity.getEvents().addListener("fertilise", this::fertiliseTile);
+        entity.getEvents().addListener("plant", this::plantCrop);
+        entity.getEvents().addListener("harvest", this::harvestCrop);
+    }
+
+    /**
+     * Decreases water content in the tile by a constant amount.
+     */
+    @Override
+    public void update() {
+        waterContent -= waterDecreaseRate * ServiceLocator.getTimeSource().getDeltaTime();
+        waterContent = waterContent < 0 ? 0 : (waterContent > 2 ? 2 : waterContent);
+    }
+
+    private void waterTile(float amount) {
```

Comment:
Should pretty much always just water to full for that day/time period so not 100% needed for amount?

Reply:
nvm read your later code. So each plant has an amount of water that is best rather than just the player instantly getting it to it's desired state and it slowly depletes over time? Having a variable amount of water each time will complicate the amount of water we need to take away from the watering can which would need an event in return to get the amount of water needed back (I really don't wanna do this). I think just a flat rate no matter what to water it so it simplifies it would be better, also having either not watered or is watered would make it easy for players to play and our code in general.

Reply:
+1. should jsut water a static constant amount

Reply:
We would be watering a flat amount each time, but we made amount a float variable because then we can take the float value in the trigger when the event is triggered, so that we can decide how much water we want to add later after discussing.

Reply:
I don't think it would be fair to get you to get the amount of water needed back to adjust the water level of the watering can. Instead, the amount of water depleted from the watering could just be constant, and the `waterLevel` of the crop tile entity would just have a max of 2 for example. For example, if a crop tile with a water level of 1.8 was hit with a water event, and the water event base amount was 0.5, the can would be depleted by 0.5 and the water level would just be updated to 2.0. I don't think that the watering system needs to be that complicated and handle edge cases of a player just spamming the watering can on a tile.

Does this make sense? Hopefully I'm not just rambling on :)

Reply:
yeh this sounds good. we were originally just worried about having to handle the water used coming back to the watering can. happy to have the watering can deplete by a constant amount on our end and yall handle the water in tile